src/main/java/uk/ac/cam/cl/dtg/util/email/MailJetApiClientWrapper.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2021 James Sharkey
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ * 		http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.cam.cl.dtg.util.email;
+
+import com.google.inject.Inject;
+import com.mailjet.client.ClientOptions;
+import com.mailjet.client.MailjetClient;
+import com.mailjet.client.MailjetRequest;
+import com.mailjet.client.MailjetResponse;
+import com.mailjet.client.errors.MailjetClientRequestException;
+import com.mailjet.client.errors.MailjetException;
+import com.mailjet.client.resource.Contact;
+import com.mailjet.client.resource.ContactManagecontactslists;
+import com.mailjet.client.resource.Contactdata;
+import com.mailjet.client.resource.Contacts;
+import com.mailjet.client.resource.ContactslistImportList;
+import org.apache.commons.lang3.Validate;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.ac.cam.cl.dtg.util.PropertiesLoader;
+
+import javax.ws.rs.core.Response;
+
+import static uk.ac.cam.cl.dtg.segue.api.Constants.*;
+
+public class MailJetApiClientWrapper {
```

Comment:
I would maybe have considered adding some kind of interface to allow us to change our mind more easily on mail provider but this could be refactored later if we think there is a chance we will want to change.

Reply:
Yes, this is a balance between baking into an interface the inefficient design decisions of MailJet's API (cannot bulk create users with IDs, cannot add users with properties except when bulk subscribing, etc, etc) or being flexible enough to be more efficient with another provider in the future.
In the same way that 5 years after we removed Mongo we're still stuck with the poor implementation decisions the old interfaces force onto us, which cannot make use of the efficiency of a relational database for doing the work under the hood.